cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,392 @@
+type LookupEventsoptions struct {
+	clusterID string
+	since     string
+	pages     int
+	cluster   *cmv1.Cluster
+	url       bool
+	raw       bool
+	all       bool
+}
+
+// RawEventDetails struct for holding raw event details
+type RawEventDetails struct {
+	EventVersion string `json:"eventVersion"`
+
+	UserIdentity struct {
+		AccountId      string `json:"accountId"`
+		Type           string `json:"type"`
+		SessionContext struct {
+			SessionIssuer struct {
+				Type     string `json:"type"`
+				UserName string `json:"userName"`
+				Arn      string `json:"arn"`
+			} `json:"sessionIssuer"`
+		} `json:"sessionContext"`
+	} `json:"userIdentity"`
+	EventRegion string `json:"awsRegion"`
+	EventId     string `json:"eventID"`
+}
+
+func newwrite_eventsCmd() *cobra.Command {
+	ops := newWrite_eventsOptions()
+	listEventsCmd := &cobra.Command{
+		Use:   "write-events",
+		Short: "Prints out all cloudtrail write events to console",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("")
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+
+		},
+	}
+	listEventsCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	listEventsCmd.Flags().StringVarP(&ops.since, "since", "", "", "Duration of lookup")
+	listEventsCmd.Flags().BoolVarP(&ops.url, "url", "u", false, "Prints link link url to specific cloudtrail event")
+	listEventsCmd.Flags().BoolVarP(&ops.raw, "raw", "r", false, "Prints Raw cloudtrail event to console")
+	listEventsCmd.Flags().BoolVarP(&ops.all, "all", "A", false, "Prints all cloudtrail events")
+	listEventsCmd.Flags().IntVar(&ops.pages, "pages", 50, "Command will display X pages of Cloud Trail logs for the cluster. Pages is set to 40 by default")
+	listEventsCmd.MarkFlagRequired("cluster-id")
+	return listEventsCmd
+}
+func newWrite_eventsOptions() *LookupEventsoptions {
+
+	return &LookupEventsoptions{}
+
+}
+
+// complete fills in necessary data for the options struct
+func (o *LookupEventsoptions) complete(cmd *cobra.Command, _ []string) error {
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return err
+	}
+
+	defer connection.Close()
+
+	cluster, err := utils.GetCluster(connection, o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	o.cluster = cluster
+	o.clusterID = cluster.ID()
+
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return errors.New("this command is only available for AWS clusters")
+	}
+
+	return nil
+}
+
+// ParseDurationToUTC parses a duration string and returns a UTC time
+func ParseDurationToUTC(input string) (time.Time, error) {
+
+	duration, err := time.ParseDuration(input)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Now().UTC().Add(-duration), nil
+}
+
+// Whoami retrieves caller identity information
+func Whoami(stsClient sts.Client) (Arn string, AccountId string, err error) {
```

Comment:
**Nitpick:** this also doesn't need to be public, same comment as for `ParseDurationToUTC`.